Node.js Client API unit test failures

Build: 03/02

There are a couple of failures on Node API unit tests on Jenkins:
 
http://jenkins.marklogic.com/view/Node%20API/job/NodeAPIUnitTests_develop_xdmp-trunk/lastCompletedBuild/testReport/

```
 document content write a language-specified XML string."before all" hook
 Stack Trace
write single document: cannot process response with 400 status
undefined

 graph operations.should run a bound SPARQL update
 Stack Trace
expected {} to have property 'marklogic.com/r1'
AssertionError: expected {} to have property 'marklogic.com/r1'
    at Proxy.Assertion.fail (node_modules/should/lib/assertion.js:228:17)
    at Proxy.prop.(anonymous function) (node_modules/should/lib/assertion.js:69:14)
    at test-basic/graphs.js:246:28
    at tryCatcher (node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (node_modules/bluebird/js/main/promise.js:510:31)
    at Promise._settlePromiseAt (node_modules/bluebird/js/main/promise.js:584:18)
    at Async._drainQueue (node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues (node_modules/bluebird/js/main/async.js:15:14)
```

--- a/lib/graphs.js
+++ b/lib/graphs.js
@@ -33,7 +33,7 @@ function uriErrorTransform(message) {
   var uri = operation.uri;
   if (typeof uri === 'string' || uri instanceof String) {
     return message+' (on graph '+uri+')';
-  } 
+  }
 
   return message;
 }
@@ -106,7 +106,7 @@ function Graphs(client) {
 
 /**
  * Provides functions read, write, merge, remove, list, or query
- * with SPARQL on triple graphs. 
+ * with SPARQL on triple graphs.
  * @namespace graphs
  */
 
@@ -117,7 +117,7 @@ function Graphs(client) {
  * @param {string} contentType - the format for the graph such as application/n-quads,
  * application/n-triples, application/rdf+json, application/rdf+xml, text/n3, text/turtle,
  * or application/vnd.marklogic.triples+xml
- * @param {string} [category] - a value from the enumeration "content|metadata|permissions" 
+ * @param {string} [category] - a value from the enumeration "content|metadata|permissions"
  * that specifies whether to retrieve the triple content of the graph (the default)
  * or the metadata permissions for the graph documents
  * @param {string|transactions.Transaction}  [txid] - a string
@@ -142,7 +142,7 @@ Graphs.prototype.read = function readGraph() {
   if (typeof testArg === 'string' || testArg instanceof String) {
     if (argLen > 1) {
       uri = testArg;
-      contentType = args[1]; 
+      contentType = args[1];
     } else {
       contentType = testArg;
     }
@@ -175,7 +175,7 @@ Graphs.prototype.read = function readGraph() {
   requestOptions.method = 'GET';
   requestOptions.headers = {
       'Accept': contentType
-  }; 
+  };
   requestOptions.path = mlutil.databaseParam(connectionParams, endpoint, '&');
 
   var operation = new Operation(
@@ -467,7 +467,7 @@ function applyGraph(action, args) {
   var requestOptions = mlutil.copyProperties(connectionParams);
   requestOptions.method = (action === 'remove') ? 'DELETE' : 'HEAD';
   requestOptions.path = mlutil.databaseParam(connectionParams, endpoint, sep);
-  
+
   var operation = new Operation(
       action+' graph', this.client, requestOptions, 'empty', 'empty'
       );
@@ -535,7 +535,7 @@ Graphs.prototype.list = function listGraphs() {
   }
 
   requestOptions.path = mlutil.databaseParam(connectionParams, endpoint, sep);
-  
+
   var operation = new Operation(
       'list graphs', this.client, requestOptions, 'empty', 'single'
       );
@@ -548,7 +548,7 @@ Graphs.prototype.list = function listGraphs() {
 };
 
 /**
- * Executes a SPARQL query against the triples for the graphs; takes a 
+ * Executes a SPARQL query against the triples for the graphs; takes a
  * configuration object with the following named parameters or, as a shortcut,
  * a contentType string, any number of default graph URIs, and a SPARQL query.
  * @method graphs#sparql
@@ -712,7 +712,7 @@ Graphs.prototype.sparql = function queryGraphSPARQL() {
   requestOptions.headers = {
       'Content-Type': contentType,
       'Accept':       acceptType
-  }; 
+  };
   requestOptions.path = mlutil.databaseParam(connectionParams, endpoint, sep);
 
   var operation = new Operation(
@@ -732,16 +732,14 @@ Graphs.prototype.sparql = function queryGraphSPARQL() {
 };
 
 /**
- * Executes a SPARQL update against the graphs; takes a configuration object with 
+ * Executes a SPARQL update against the graphs; takes a configuration object with
  * the following named parameters or, as a shortcut, the SPARQL update data.
  * @method graphs#sparqlUpdate
  * @param {string|ReadableStream} data - the SPARQL update
  * @param {object[]} [permissions] - the permissions controlling which users can read or
  * write the documents with the updated triples
- * @param {boolean} [usingDefaultGraphs] - whether to use default graphs for the triples
- * in the update
- * @param {boolean} [usingNamedGraphs] - whether to use named graphs for the triples
- * in the update
+ * @param {string|string[]} [usingDefaultGraphs] - the default graphs for the update
+ * @param {string|string[]} [usingNamedGraphs] - the named graphs for the update
  * @param {string} [base] - the base URI to prepend to relative resource identifiers
  * @param {string|transactions.Transaction}  [txid] - a string
  * transaction id or Transaction object identifying an open
@@ -799,20 +797,26 @@ Graphs.prototype.sparqlUpdate = function updateGraphSPARQL() {
   var endpoint = '/v1/graphs/sparql';
   var sep = '?';
 
-  if (typeof usingDefaultGraphs === 'boolean') {
-    endpoint += sep+'using-graph-uri='+String(usingDefaultGraphs);
-    if (sep === '?') { sep = '&'; }
-  } else if (usingDefaultGraphs instanceof Boolean) {
-    endpoint += sep+'using-graph-uri='+usingDefaultGraphs.toString();
-    if (sep === '?') { sep = '&'; }
+  var hasUsingDefaultGraphs = (usingDefaultGraphs !== null && usingDefaultGraphs !== void 0);
+  if (hasUsingDefaultGraphs) {
+    usingDefaultGraphs = mlutil.asArray(usingDefaultGraphs);
+    if (usingDefaultGraphs.length > 0) {
+      endpoint += sep+encodeParamValues('using-graph-uri', usingDefaultGraphs);
+      sep = '&';
+    } else {
+      hasUsingDefaultGraphs = false;
+    }
   }
 
-  if (typeof usingNamedGraphs === 'boolean') {
-    endpoint += sep+'using-named-graph-uri='+String(usingNamedGraphs);
-    if (sep === '?') { sep = '&'; }
-  } else if (usingNamedGraphs instanceof Boolean) {
-    endpoint += sep+'using-named-graph-uri='+usingNamedGraphs.toString();
-    if (sep === '?') { sep = '&'; }
+  var hasUsingNamedGraphs = (usingNamedGraphs !== null && usingNamedGraphs !== void 0);
+  if (hasUsingNamedGraphs) {
+    usingNamedGraphs = mlutil.asArray(usingNamedGraphs);
+    if (usingNamedGraphs.length > 0) {
+      endpoint += sep+encodeParamValues('using-named-graph-uri', usingNamedGraphs);
+      if (sep === '?') { sep = '&'; }
+    } else {
+      hasUsingNamedGraphs = false;
+    }
   }
 
   var permissionsParams = makePermissionsParams(permissions, sep);
@@ -834,7 +838,7 @@ Graphs.prototype.sparqlUpdate = function updateGraphSPARQL() {
   requestOptions.method = 'POST';
   requestOptions.headers = {
       'Content-Type': 'application/sparql-update'
-  }; 
+  };
   requestOptions.path = mlutil.databaseParam(connectionParams, endpoint, sep);
 
   var operation = new Operation(

--- a/test-basic/graphs.js
+++ b/test-basic/graphs.js
@@ -229,7 +229,8 @@ describe('graph operations', function(){
   it('should run a bound SPARQL update', function(done) {
     this.timeout(3000);
     db.graphs.sparqlUpdate({
-        usingNamedGraphs: true,
+        usingDefaultGraphs: graph3Uri, // for coverage, superfluous given the query
+        usingNamedGraphs: [graph2Uri, graph3Uri], // for coverage, superfluous given the query
         data:
           'CREATE GRAPH <'+graph3Uri+'> ; '+
           'INSERT { GRAPH <'+graph3Uri+'> { ?s ?p "literal2" } } '+

